src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+
+  const filteredMen = century
+    ? men.filter(man => Math.ceil(man.died / 100) === century)
+    : men;
+
+  const averageAge = filteredMen.reduce((sum, man) =>
+    sum + (man.died - man.born), 0) / filteredMen.length || 0;
```

Comment:
Create separate function `calculateAverageAge` and move this logic to this function.
Now you have the logic with reduce repeated in all 3 functions.
Make it so that the function takes an array of numbers and the length by which it needs to be divided